Sources/Subprocess/IO/AsyncIO+Windows.swift

```diff
@@ -0,0 +1,437 @@
+                // Read some data
+                readLength += Int(bytesRead)
+                if maxLength == .max {
+                    // Grow resultBuffer if needed
+                    guard Double(readLength) > 0.8 * Double(resultBuffer.count) else {
+                        continue
+                    }
+                    resultBuffer.append(
+                        contentsOf: Array(repeating: 0, count: resultBuffer.count)
+                    )
+                } else if readLength >= maxLength {
+                    // When we reached maxLength, return!
+                    return resultBuffer
+                }
+            }
+        }
+    }
+
+    func write(
+        _ array: [UInt8],
+        to diskIO: borrowing IOChannel
+    ) async throws -> Int {
+        return try await self._write(array, to: diskIO)
+    }
+
+#if SubprocessSpan
+    func write(
+        _ span: borrowing RawSpan,
+        to diskIO: borrowing IOChannel
+    ) async throws -> Int {
+        let handle = diskIO.channel
+        var signalStream = self.registerHandle(diskIO.channel).makeAsyncIterator()
+        var writtenLength: Int = 0
+        while true {
+            var overlapped = _OVERLAPPED()
+            let succeed = try span.withUnsafeBytes { ptr in
+                // Windows WriteFile uses DWORD for target count
+                // which means we can only write up to DWORD max
+                let remainingLength: DWORD
+                if MemoryLayout<Int>.size == MemoryLayout<Int32>.size {
+                    // On 32 bit systems we don't have to worry about overflowing
+                    remainingLength = DWORD(truncatingIfNeeded: ptr.count - writtenLength)
+                } else {
+                    // On 64 bit systems we need to cap the count at DWORD max
+                    remainingLength = DWORD(truncatingIfNeeded: min(ptr.count - writtenLength, Int(DWORD.max)))
+                }
+
+                let startPtr = ptr.baseAddress!.advanced(by: writtenLength)
+                return WriteFile(
+                    handle,
+                    startPtr,
+                    DWORD(truncatingIfNeeded: remainingLength),
+                    nil,
+                    &overlapped
+                )
+            }
+            if !succeed {
+                // It is expected `WriteFile` to return `false` in async mode.
+                // Make sure we only get `ERROR_IO_PENDING`
+                let lastError = GetLastError()
+                guard lastError == ERROR_IO_PENDING else {
+                    let error = SubprocessError(
+                        code: .init(.failedToWriteToSubprocess),
+                        underlyingError: .init(rawValue: lastError)
+                    )
+                    throw error
+                }
+
+            }
+            // Now wait for read to finish
+            let bytesWritten: DWORD = try await signalStream.next() ?? 0
+
+            writtenLength += Int(bytesWritten)
```

Comment:
Here's another place where a DWORD (UInt32) to Int conversion could crash on 32-bit Windows with large-enough values